src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private StorageDao storageDao;
+
+    public PurchaseOperationHandler() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Override
+    public Map<String, Integer> getOperation(FruitTransaction transaction) {
+        storageDao.save(transaction.getFruit(),
+                storageDao.getQuantity(transaction.getFruit()) - transaction.getQuantity());
```

Comment:
The method does not handle cases where the purchase quantity exceeds the available quantity, which could lead to negative values in the storage. Consider adding a check to prevent this scenario.